Quick-Start [DA]/MVP/MVP+Makefile.md

```diff
@@ -1 +1,274 @@
+
+# [CODE](https://github.com/IshaanAdarsh/Postgres-extension-tutorial/tree/main/Code/my_extension)
+
+## A Postgres extension using only a control file, an SQL file and a Makefile:
+
+### Step 1: Create the Extension Directory
+- Create a directory for your extension. Let's call it "my_extension".
+
+```bash
+$ mkdir my_extension
+$ cd my_extension
+```
+
+### Step 2: Create the Extension Control File
+- Inside the "my_extension" directory, create a file named "my_extension.control". 
+  - This file defines the metadata and dependencies of your extension.
+
+```bash
+$ touch my_extension.control
+```
+
+- Open the "my_extension.control" file and add the following content:
+
+```control
+# my_extension.control
+comment = 'Minimal Viable Product'
+default_version = '1.0'
+relocatable = true
+
+# Installation script for version 1.0
+# 1.0
+# my_extension--1.0.sql
+```
+
+### Step 3: Create the Extension SQL File
+- Create a file named "my_extension--1.0.sql".
+  - This file contains the SQL statements required to install or uninstall your extension.
+
+```bash
+$ touch my_extension--1.0.sql
+```
+
+- Open the "my_extension--1.0.sql" file and add the following content:
+
+```sql
+-- extension--1.0.sql
+-- my_extension--1.0.sql
+-- Create necessary objects for version 1.0
+CREATE TABLE my_table (
+  id SERIAL PRIMARY KEY,
+  name VARCHAR(100) NOT NULL
+);
+```
+
+### Step 4: Create the Makefile
+- Create a Makefile in the "extension" directory. 
+  - The Makefile is used to define the build process. It specifies the module name, object files, extension name, and SQL files. The PG_CONFIG variable retrieves the path to the pg_config utility, which is used to get the necessary configuration for building the extension.
+
+```bash
+$ touch Makefile
+```
+- Open the Makefile and add the following content:
+<!--Problem: https://dba.stackexchange.com/questions/328624/makefile-issue-in-creating-postgres-extension
+Had an issue with the make install command:
+```makefile
+DATA = my_extension--1.0.sql
+
+PG_CONFIG  ?= pg_config
+PGXS := $(shell $(PG_CONFIG) --pgxs)
+include $(PGXS)
+```
+Solution: Makefile is lacking the entry `EXTENSION = my_extension`
+Reason: That's why `make install` doesn't copy the `*.control` file, and that's why `CREATE EXTENSION` fails.
+
+NOTE: make: Nothing to be done for 'all' 
+Makefile doesn't have any MODULE or other binary to build. You can go directly to the make install step
+-->
+
+```makefile
+EXTENSION = my_extension
+DATA = my_extension--1.0.sql
+
+PG_CONFIG  ?= pg_config
+PGXS := $(shell $(PG_CONFIG) --pgxs)
+include $(PGXS)
+```
+
+### Step 5: Build and Install the Extension
+- To build and install the extension, run the following commands:
+
+```bash
+$ make
+$ make install
+```
+- The "make" command compiles the extension code using the Makefile
```

Comment:
This is true, although here there is nothing to compile.

Reply:
yup I mentioned this in the comments, I will add a line that specifies in this particular extension code there is No Modules to install, so make doesn't serve a purpose.